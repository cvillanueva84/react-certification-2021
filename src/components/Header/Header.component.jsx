import React from 'react';
import styled from 'styled-components';
import avatar from '../../assets/images/undraw_female_avatar.svg';
import logoDesktop from '../../assets/images/logo-desktop.png';
import logoMobile from '../../assets/images/mobile-icon.png';
import ToggleSwitch from '../ToggleSwitch';

const HeaderStyled = styled.header`
  background-color: #ffffff;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0.75rem 2rem;
  position: fixed;
  top: 0;
  width: 100%;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
  .margin-x {
    margin: 0 0.75rem;
  }
  .menu-hamburger {
    display: none;
  }
  @media (max-width: 576px) {
    padding: 1rem 0.75rem;

    .hide-mobile {
      display: none;
    }

    .menu-hamburger {
      display: block;
    }
  }
`;

const InputSearch = styled.input`
  outline: none;
  width: 100%;
  padding: 12px 20px;
  margin: 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
  &:focus {
    border-color: #928a97;
  }
  @media (max-width: 576px) {
    padding: 10px;
  }
`;

const Avatar = styled.img`
  height: 50px;
  width: 50px;
  @media (max-width: 576px) {
    height: 40px;
    width: 40px;
  }
`;

const Logo = styled.img`
  height: 35px;
`;

const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;

const Header = () => (
  <HeaderStyled>
    <Wrapper>
      <picture>
        <source srcSet={logoMobile} media="(max-width: 576px)" />
        <Logo src={logoDesktop} alt="Pied Piper logo" className="margin-x" />
      </picture>
      <form className="margin-x">
        <InputSearch type="text" placeholder="Search" />
      </form>
    </Wrapper>
    <Wrapper className="nav hide-mobile">
      <div className="margin-x hide-mobile">
        <ToggleSwitch />
      </div>
      <figure className="margin-x">
        <Avatar src={avatar} alt="current user avatar" />
      </figure>
    </Wrapper>
  </HeaderStyled>
);

export default Header;
