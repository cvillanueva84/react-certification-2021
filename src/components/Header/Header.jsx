import React, { useState, useContext } from 'react';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import Switch from '@material-ui/core/Switch';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { AppContext } from '../../providers/App/App.provider';
import { primary, dark } from '../../style/theme';
import {
  Container,
  LeftHeader,
  RightHeader,
  SearchDiv,
  DivSearchIcon,
  DivInput,
  ButtonAccount,
  DivAvatar,
} from './Header.styled';

function Header({ children, enterPressed }) {
  const [checked, setChecked] = useState(false);
  const { state, dispatch } = useContext(AppContext);
  const toggleChecked = () => {
    setChecked((prev) => !prev);
    if (checked) {
      dispatch({ type: 'SET_THEME', payload: { theme: primary } });
    } else {
      dispatch({ type: 'SET_THEME', payload: { theme: dark } });
    }
  };
  return (
    <Container theme={state.theme}>
      <LeftHeader>
        <MenuIcon />
        <SearchDiv type="submit" onSubmit={enterPressed}>
          <DivInput>
            <DivSearchIcon>
              <SearchIcon />
            </DivSearchIcon>
            {children}
          </DivInput>
        </SearchDiv>
      </LeftHeader>
      <RightHeader>
        <div>
          <FormGroup>
            <FormControlLabel
              control={<Switch checked={checked} onChange={toggleChecked} />}
              label="Dark Mode"
            />
          </FormGroup>
        </div>
        <ButtonAccount>
          <DivAvatar>
            <AccountCircleIcon style={{ fontSize: 40 }} />
          </DivAvatar>
        </ButtonAccount>
      </RightHeader>
    </Container>
  );
}

export default Header;
