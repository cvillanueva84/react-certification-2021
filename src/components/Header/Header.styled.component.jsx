/* eslint-disable import/no-extraneous-dependencies */
import styled from 'styled-components';
import { Menu } from '@styled-icons/boxicons-regular/Menu';

export const HeaderContainer = styled.header`
  background-color: #1c5476;
  color: white;
  min-height: 64px;
  width: 100%;
  display: flex;
  align-self: flex-between;
  padding: 0 24px;
`;

export const LeftHeaderContainer = styled.div`
  display: flex;
  justify-content: start;
  width: 100%;
`;

export const BurgerMenuContainer = styled.div`
  /* padding: 12px 20px; */
  display: flex;
  justify-content: center;
  min-width: 48px;
`;

export const BurgerMenu = styled(Menu)`
  color: ${(props) => props.color};
  margin-right: 16px;
  padding: 12px 0;
`;
