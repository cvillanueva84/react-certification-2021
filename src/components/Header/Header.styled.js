import styled from 'styled-components';

const HeaderWrapper = styled.div`
  background-color: transparent;
  height: 60px;
  display: flex;
  width: 100%;
  align-items: center;
  padding: 10px 20px;
  justify-content: space-between;
  border-bottom: 1px solid #dcdcdc;
`;

const FlexCentered = styled.div`
  display: flex;
  align-items: center;
`;

const Input = styled.input`
  padding: 10px;
  width: 400px;
  border: 1px solid #8e8e8e;
  height: 41px;
`;

const Button = styled.button`
  padding: 10px;
  background: none;
  border: 1px solid #8e8e8e;
  width: 70px;
  background-color: #8e8e8e;
  height: 41px;
`;

export { Button, Input, FlexCentered, HeaderWrapper };
