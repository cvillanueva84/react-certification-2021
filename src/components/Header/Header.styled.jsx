import styled from 'styled-components';

export const Container = styled.header`
  color: white;
  width: 100%;
  height: 64px;
  display: flex;
  background-color: ${(props) => props.theme['background-header']};
  justify-content: space-between;
  align-items: center;
`;
export const LeftHeader = styled.div`
  display: flex;
  align-items: center;
  margin-left: 30px;
`;
export const RightHeader = styled.div`
  display: flex;
  align-items: center;
  margin-right: 30px;
  @media (max-width: 768px) {
    display: none;
  }
`;

export const SearchDiv = styled.form`
  width: auto;
  margin-left: 24px;
  position: relative;
  margin-right: 16px;
  border-radius: 4px;
  background-color: rgba(255, 255, 255, 0.2);
`;
export const DivSearchIcon = styled.button`
  display: flex;
  color: white;
  height: 100%;
  padding: 0px 16px;
  position: absolute;
  align-items: center;
  pointer-events: none;
  justify-content: center;
  background-color: transparent;
  z-index: 1;
  border: none;
`;

export const DivInput = styled.div`
  display: inline-flex;
  position: relative;
  align-items: center;
`;

export const SwitchBtn = styled.span`
  width: 58px;
  height: 38px;
  display: inline-flex;
  padding: 12px;
  z-index: 0;
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  flex-shrink: 0;
  vertical-align: middle;
`;

export const IconButton = styled.span`
  top: 0;
  left: 0;
  color: #fafafa;
  z-index: 1;
  position: absolute;
  transition: left 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
    transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 9px;
  overflow: visible;
  font-size: 1.5rem;
  text-align: center;
  border-radius: 50%;
  border: 0;
  margin: 0;
  display: inline-flex;
  outline: 0;
  align-items: center;
  user-select: none;
  vertical-align: middle;
  justify-content: center;
  text-decoration: none;
  background-color: transparent;
`;

export const LabelIconButton = styled.span`
  width: 100%;
  display: flex;
  color: white;
  font-size: 1.5rem;
  text-align: center;
  align-items: inherit;
  justify-content: inherit;
`;

export const Input = styled.input`
  left: -100%;
  width: 300%;
  top: 0;
  height: 100%;
  z-index: 1;
  position: absolute;
  padding: 0;
  opacity: 0;
  cursor: inherit;
  margin: 0;
`;

export const Thumb = styled.span`
  color: white;
  width: 20px;
  height: 20px;
  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%),
    0px 1px 3px 0px rgb(0 0 0 / 12%);
  border-radius: 50%;
  background-color: currentColor;
`;

export const TouchRipple = styled.span`
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 0;
  overflow: hidden;
  position: absolute;
  border-radius: inherit;
  pointer-events: none;
`;

export const TrackSwitch = styled.span`
  width: 100%;
  height: 100%;
  opacity: 0.38;
  z-index: -1;
  transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
    background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 7px;
  background-color: #000;
`;

export const ButtonAccount = styled.button`
  color: inherit;
  height: 100%;
  padding: 12px;
  margin-left: 12px;
  align-items: center;
  cursor: pointer;
  outline: 0;
  background-color: transparent;
  border: 0;
`;

export const DivAvatar = styled.div`
  width: 40px;
  height: 40px;
  overflow: hidden;
  align-items: center;
`;
