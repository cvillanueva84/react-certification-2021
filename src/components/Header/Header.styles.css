/***************MENU TOGGLE****************/
.header-icon span {
  cursor: pointer;
}

/***************SEARCH BAR****************/
.search-bar {
  align-items: center;
  display: flex;
}

.search-bar input,
.search-bar button {
  border: none;
  height: 2.5em;
}

.search-bar input {
  background-color: #c2c3ff;
  border-radius: 4px 0 0 4px;
  color: #191a5e;
  width: 100%;
}

.search-bar button {
  background-color: #191a5e;
  border-radius: 0 4px 4px 0;
  cursor: pointer;
}

/***************PROFILE MENU****************/
.input-icon {
  background-color: #c2c3ff;
  border-radius: 50%;
  cursor: pointer;
  height: 2em;
  width: 2em;
}
