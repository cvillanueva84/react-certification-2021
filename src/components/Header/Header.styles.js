import styled from "styled-components";
import hamburguer from '../../img/menu.png';
import profile from '../../img/user.png';

export const HeaderBody = styled.div`
    display: flex;
    align-items: center;
    text-align: center;
`;

export const HeaderContainer = styled.div`
    display: flex;
    justify-content: space-between;
    width: 100vw;
    min-height: 60px;
    background: #1C5476;
    padding: 15px 20px;
    color: #fff;
    display: flex;
    box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%), 0px 4px 5px 0px rgb(0 0 0 / 14%), 0px 1px 10px 0px rgb(0 0 0 / 12%);
`;

export const Hamburguer = styled.span`
    background-image: url(${hamburguer});
    width: 30px;
    height: 30px;
    background-size: contain;
    background-repeat: no-repeat;
`;

export const Search = styled.input`
    padding: 10px;
    margin: 5px 0px;
    background: white;
    border: none;
    border-radius: 3px;
    height: 30px;
    margin-left: 30px;
    background-color: #3E6D8A;
    color: white;
    
    ::placeholder {
        color: white;
    }

    @media (max-width: 485px) {
        margin-left: 15px;
    }
    @media (max-width: 768px) {
        
    }
`;

export const ToggleBox = styled.input`    
    border-radius: 15px;
    width: 42px;
    height: 26px;
    margin-right: 30px;

    @media (max-width: 485px) {
        margin-right: 15px;
    }
    @media (max-width: 768px) {
        
    }
`;

export const Profile = styled.span`
    background-image: url(${profile});
    width: 30px;
    height: 30px;
    background-size: contain;
    background-repeat: no-repeat;
`;