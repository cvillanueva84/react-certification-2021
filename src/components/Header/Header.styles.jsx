import styled from 'styled-components';

const HeaderBody = styled.div`
  border-bottom: solid 1px #f1f1f1;
  display: flex;
  justify-content: space-between;
  padding: 0.75rem;
  @media (max-width: 1000px) {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(25rem, 1fr));
    gap: 0.5rem;
  }
`;
const HeaderLeft = styled.div`
  display: flex;
  gap: 2rem;
`;
const HeaderItems = styled.div`
  display: flex;
  gap: 2rem;
  @media (max-width: 1000px) {
    display: none;
  }
`;
const Search = styled.input`
  padding: 16px 16px 16px 60px;
  border: 1px solid;
  border-radius: 3px;
  box-shadow: inset 0 -8px 1px rgba(248, 10, 10, 0.03);
  font-weight: 300;
  font-size: 15px;
  width: 400px;
  height: 30px;
  :hover {
    background-color: #f8f7f7;
  }
  ::placeholder,
  ::-webkit-input-placeholder {
    color: #dbd8d8;
    font-weight: 300;
    font-size: 1.1rem;
  }
`;
const Button = styled.button`
  font-size: 1em;
  font-weight: 600;
  padding: 0.1rem 1rem;
  text-align: center;
  border-radius: 50%;
  background-image: url(${(props) => props.avatarUrl});
  background-position: center;
  background-size: 60px 50px;
  background-repeat: no-repeat;
  :hover {
    border: 2px solid black;
  }
`;
const CheckBoxWrapper = styled.div`
  position: relative;
  top: 10px;
`;
const CheckBoxLabel = styled.label`
  position: absolute;
  left: 0;
  width: 42px;
  border-radius: 15px;
  background: #bebebe;
  &::after {
    content: '';
    display: block;
    border-radius: 50%;
    width: 18px;
    height: 18px;
    margin: 3px;
    background: #ffffff;
    box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.2);
    transition: 0.2s;
  }
`;
const CheckBox = styled.input`
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
  width: 42px;
  height: 26px;
  &:checked + ${CheckBoxLabel} {
    background: red;
    &::after {
      content: '';
      display: flex;
      border-radius: 50%;
      width: 18px;
      height: 18px;
      margin-left: 21px;
      transition: 0.2s;
    }
  }
`;

export {
  HeaderBody,
  HeaderLeft,
  HeaderItems,
  Search,
  Button,
  CheckBoxWrapper,
  CheckBoxLabel,
  CheckBox,
};
