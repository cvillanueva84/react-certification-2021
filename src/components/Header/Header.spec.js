import Header from './Header.component';

const { render, screen } = require('@testing-library/react');

describe('Header', () => {
  let container;

  beforeEach(() => {
    container = render(<Header />).container;
  });

  test('Should renders Header component', () => {
    expect(container).toMatchSnapshot();
  });

  test('Should contains a logo image', () => {
    const imageElement = screen.getByAltText(/pied piper logo/i);
    expect(imageElement).toBeInTheDocument();
  });

  test('Should contains an user avatar', () => {
    const imageElement = screen.getByAltText(/current user avatar/i);
    expect(imageElement).toBeInTheDocument();
  });

  test('Should contains a toggle switch button', () => {
    const togleSwitchButton = screen.getByRole('switch');
    expect(togleSwitchButton).toBeInTheDocument();
    expect(togleSwitchButton.children.length).toBe(2);
    expect(togleSwitchButton.firstChild).toHaveAttribute('type', 'checkbox');
  });

  test('Should contains an input', () => {
    const inputElement = screen.getByPlaceholderText(/search/i);
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('type', 'text');
  });
});
