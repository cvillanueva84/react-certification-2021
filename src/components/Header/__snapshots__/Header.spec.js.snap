// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should renders Header component 1`] = `
<div>
  <header
    class="sc-hKFxyN htvvhm"
  >
    <div
      class="sc-iCoGMd ihqXQh"
    >
      <picture>
        <source
          media="(max-width: 576px)"
          srcset="mobile-icon.png"
        />
        <img
          alt="Pied Piper logo"
          class="sc-gKAaRy eWqJQX margin-x"
          src="logo-desktop.png"
        />
      </picture>
      <form
        class="margin-x"
      >
        <input
          class="sc-eCApnc dIlmAH"
          placeholder="Search"
          type="text"
        />
      </form>
    </div>
    <div
      class="sc-iCoGMd ihqXQh nav hide-mobile"
    >
      <div
        class="margin-x hide-mobile"
      >
        <label
          class="sc-bdnxRM fzaqfv switch"
          for="switch-dark-mode"
          role="switch"
        >
          <input
            class="sc-gtsrHT flAbm"
            id="switch-dark-mode"
            role="button"
            type="checkbox"
          />
          <span
            class="sc-dlnjwi dSfqLy"
          />
        </label>
      </div>
      <figure
        class="margin-x"
      >
        <img
          alt="current user avatar"
          class="sc-jSFjdj elchRP"
          src="undraw_female_avatar.svg"
        />
      </figure>
    </div>
  </header>
</div>
`;
